src/Workspaces/Core/Portable/Execution/AssetStorages.Storage.cs

```diff
@@ -24,6 +24,7 @@ internal partial class AssetStorages
         public sealed class Storage
         {
             private readonly ISerializerService _serializer;
+            private readonly List<object> _keepAliveObjects = new List<object>();
```

Comment:
any threading concerns?

Reply:
The writer is inherently single-threaded. Are multiple writers for a single callback allowed concurrently?

Reply:
➡️ Added a lock